src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,52 +1,147 @@
 package core.basesyntax;
 
-/**
- * <p>Реалізувати свій ArrayList який імплементує інтерфейс List</p>
- */
 public class ArrayList<T> implements List<T> {
+    private int size;
+    private Object[] array = new Object[10];
+    private Object[] tempArray = new Object[10];
 
     @Override
     public void add(T value) {
-
+        if (size >= array.length) {
+            tempArray = new Object[array.length];
```

Comment:
try to use method System.arraucopy()
and better move this logic in separate private method witch will increase size of the array